src/main/java/com/vmware/vipclient/i18n/messages/api/opt/server/StringBasedOpt.java

```diff
@@ -105,4 +115,51 @@ public String getTranslationStatus() {
         }
         return status;
     }
+
+    public void getMultiVersionKeyMessages(MessageCacheItem cacheItem) {
+        String url = V2URL.getMultiVersionKeyTranslationURL(this.dto,
+                VIPCfg.getInstance().getVipService().getHttpRequester().getBaseURL());
+
+        Map<String, String> headers = new HashMap<String, String>();
+        if (cacheItem.getEtag() != null)
+            headers.put(URLUtils.IF_NONE_MATCH_HEADER, cacheItem.getEtag());
+
+        Map<String, Object> response = VIPCfg.getInstance().getVipService().getHttpRequester()
+                .request(url, ConstantsKeys.GET,null, headers);
+
+        Integer responseCode = (Integer) response.get(URLUtils.RESPONSE_CODE);
+
+        if (responseCode != null && (responseCode.equals(HttpURLConnection.HTTP_OK) ||
+                responseCode.equals(HttpURLConnection.HTTP_NOT_MODIFIED))) {
```

Comment:
if HTTP_NOT_MODIFIED, why still need to request the data?

Reply:
HTTP_NOT_MODIFIED is http response code, client won't calculate the hash of the parameters and then decide send request or not. For each request, client will send it to Singleton service, Singleton service return HTTP_OK or HTTP_NOT_MODIFIED, if HTTP_NOT_MODIFIED it won't return the no change data.